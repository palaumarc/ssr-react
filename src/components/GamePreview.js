import React from "react";
import PropTypes from 'prop-types';

const contentDivStyles = {
    padding: '5px',
    height: '250px',
    border: '1px solid grey',
    display: 'grid',
    gridTemplateRows: '80% 20%',
    placeItems: 'center'
};

const logoImgStyles = {
    width: '100%',
    height: '100%',
    objectFit: 'contain'
}

const GamePreview = ({ name, logoUrl }) => (
    <div style={contentDivStyles} >
        <img style={logoImgStyles} alt={name} src={logoUrl} />
        <span>{ name }</span>
    </div>
)

GamePreview.propTypes = {
    name: PropTypes.string,
    logoUrl: PropTypes.string,
};


export default GamePreview;