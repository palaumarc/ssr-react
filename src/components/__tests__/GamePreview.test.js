import React from 'react';
import GamePreview from '../GamePreview';
import renderer from 'react-test-renderer';

it('All props are defined. Snapshot', () => {
    const name = 'dummyName';
    const logoUrl = 'dummyLogoUrl'
    const tree = renderer.create(<GamePreview name={name} logoUrl={logoUrl} />).toJSON();
    expect(tree).toMatchSnapshot();
});