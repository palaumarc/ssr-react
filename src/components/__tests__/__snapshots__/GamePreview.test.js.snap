// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All props are defined. Snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid grey",
      "display": "grid",
      "gridTemplateRows": "80% 20%",
      "height": "250px",
      "padding": "5px",
      "placeItems": "center",
    }
  }
>
  <img
    alt="dummyName"
    src="dummyLogoUrl"
    style={
      Object {
        "height": "100%",
        "objectFit": "contain",
        "width": "100%",
      }
    }
  />
  <span>
    dummyName
  </span>
</div>
`;
